src/main/java/org/jenkins/tools/test/model/MavenPom.java

```diff
@@ -229,13 +230,13 @@ public void addDependencies(
             Map<String, VersionNumber> toReplaceTest,
             Map<String, String> pluginGroupIds,
             List<String> toConvert)
-            throws IOException {
+            throws PomTransformationException {
         File pom = new File(rootDir.getAbsolutePath() + "/" + pomFileName);
         Document doc;
         try {
             doc = new SAXReader().read(pom);
         } catch (DocumentException x) {
-            throw new IOException(x);
+            throw new PomTransformationException("Failed to parse pom.xml", x);
```

Comment:
Making this class consistent with the error handling philosophy elsewhere of failing fast and avoiding abuse of `IOException`.